include/kframework/foundry.md

```diff
@@ -246,7 +266,8 @@ First we have some helpers in K which can:
 ```k
     syntax KItem ::= #assume ( Bool ) [klabel(foundry_assume), symbol]
  // ------------------------------------------------------------------
-    rule <k> #assume(B) => . ... </k> ensures B
+    rule <k> #assume(B) ~> _ => #halt </k> <statusCode> _ => FOUNDRY_VACUOUSSUCCESS </statusCode> requires notBool B
+    rule <k> #assume(B) => . ... </k> requires B
```

Comment:
This is going to produce a branch every time there is an `assume`, right? I think this has some undesirable consequences:

1. It will impact performance somewhat. It's probably not a huge impact because one branch will be vacuous and will halt, but it means the overhead of branching and saving to the kcfg for every `assume` in the code.
2. It will make kcfgs harder to read, because instead of a linear execution we will have multiple branching points, and each branching point indents the kcfg further to the right.
3. It will probably be confusing to users, because there will be branching that they won't expect. Also, this means that every code with `assume` will have vacuous branches, which might make it harder for the user to notice when actual vacuous branches show up.

What would happen if we instead had 
```
rule <k> #assume(false) ~> _ => #halt </k> <statusCode> _ => FOUNDRY_VACUOUSSUCCESS </statusCode>
rule <k> #assume(B) => . ... </k> ensures B
```
Would this still cause branching, or does the fact that `ensures false` is not allowed prevent it? Are there other downsides?

If we end up keeping the current version, I think it would be best to put it behind a command-line option like `--allow-vacuous-assume`, so users don't have to deal with it unless they know they need it.

Reply:
I didn't see @ehildenb's comment before writing this, but yeah, I think we agree on this.

Reply:
Yes, having two rules like the ones you suggested would still cause branching. So I either keep the semantic changes and add a flag to enable/disable this behaviour or I take Everett's suggestion and do the check on the pyk side.